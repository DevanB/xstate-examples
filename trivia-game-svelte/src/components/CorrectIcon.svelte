<div class="correctIcon">
  <svg viewBox='0 0 20 20'>
    <path d='M0 11l2-2 5 5L18 3l2 2L7 18z' fill='currentColor' />
  </svg>
</div>

<style>
.correctIcon {
  @apply h-4 w-4 text-green-500;
}

.correctIcon:hover {
  @apply text-green-700;
}
</style>
