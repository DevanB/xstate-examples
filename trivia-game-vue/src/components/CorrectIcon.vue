<template>
  <p>correct</p>
</template>

<script>
export default {
  name: "CorrectIcon"
}
</script>
