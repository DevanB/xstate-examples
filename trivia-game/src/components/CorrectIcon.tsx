import React from 'react'
import tw from 'tailwind.macro'

const CheckmarkSvg = () => (
  <svg viewBox='0 0 20 20'>
    <path d='M0 11l2-2 5 5L18 3l2 2L7 18z' fill='currentColor' />
  </svg>
)

const Icon = tw.div`h-4 w-4 text-green-500 hover:text-green-700`

export const CorrectIcon = () => (
  <Icon>
    <CheckmarkSvg />
  </Icon>
)
