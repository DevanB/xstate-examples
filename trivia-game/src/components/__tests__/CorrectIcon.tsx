import {render} from '@testing-library/react'
import React from 'react'
import {CorrectIcon} from '../CorrectIcon'

describe('<CorrectIcon/>', () => {
  it('should match snapshot', () => {
    const {container} = render(<CorrectIcon />)
    expect(container).toMatchSnapshot()
  })
})
