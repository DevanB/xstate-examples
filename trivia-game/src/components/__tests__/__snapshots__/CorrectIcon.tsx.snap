// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CorrectIcon/> should match snapshot 1`] = `
<div>
  <div
    class="CorrectIcon__Icon-yoxooe-0 jRabTQ"
  >
    <svg
      viewBox="0 0 20 20"
    >
      <path
        d="M0 11l2-2 5 5L18 3l2 2L7 18z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
